agent/eventhandler/task_handler.go

```diff
@@ -387,6 +411,33 @@ func (taskEvents *taskSendableEvents) sendChange(change *sendableEvent,
 		// If a send event is not already in progress, trigger the
 		// submitTaskEvents to start sending changes to ECS
 		taskEvents.sending = true
+
+		//If network connection is successfully up, keep track of how many
+		// task events are being sent in order to avoid hitting the throttlingLimit.
+		if cfg.DisconnectCapable.Enabled() && !cfg.GetDisconnectModeEnabled() {
+			if handler.taskCount == 0 {
+				logger.Debug("Starting taskCountTimer here.")
+				handler.taskCountTimer = time.NewTimer(time.Duration(throttlingTimeOut))
+			}
+			handler.taskCount++
+			logger.Debug("Increasing taskCount by 1", logger.Fields{
```

Comment:
The log setup here seems to be excessive for me for debug log level (we are expecting to see 120 line of logs which just says increase by 1, maybe remove it?

Reply:
Makes sense, I will remove!